src/sql/store.py

```diff
@@ -77,8 +77,12 @@ def store(self, key, query, with_=None):
             )
         if with_ and key in with_:
             raise ValueError(f"Script name ({key!r}) cannot appear in with_ argument")
+        # Remove trailing semicolon from query
+        query_without_trailing_semicolon = query
+        if query.strip().endswith(";"):
+            query_without_trailing_semicolon = query[::-1].replace(";", "", 1)[::-1]
```

Comment:
this logic seems duplicated, do we need it? I think as long as we have it in the SQLQuery.__str__ method, it should be fine

Reply:
when you run the save method by itself (without a with), i don't believe that SQLQuery.str gets called, so without the logic here, a trailing semicolon in a single save statement wont be removed when its stored

Reply:
tha's fine, the problem is when rendering the CTE, when storing it, we can keep the `;`

Reply:
if we take this out, there needs to be extra logic to clean the saved queries when referring to them in the render method, specifically every single `saved[name]._query`:

```
with_clause_template = Template(
            """WITH{% for name in with_ %} {{name}} AS ({{saved[name]._query}})\
{{ "," if not loop.last }}{% endfor %}{{query}}"""
        )
```

Reply:
sorry for the delay.

I see what you mean. The problem is that some SQL engines *require* a `;` at the end of the query, so this might break some queries:

```
%%sql --save
select * from table; -- this will break if the db requires a ";" at the end
```

I think the solution is to move the semicoling removing logic into a function and then call it from the templating code:

```
with_clause_template = Template( """WITH{% for name in with_ %} {{name}} AS ({{remove_semicolon(saved[name]._query)}})\ {{ "," if not loop.last }}{% endfor %}{{query}}""" )
```

in jinja, you can pass functions in the `render` method so it should be straightforward to fix


this way we only modify the queries when we have to (when generating the CTE)


let me know if you still have time to fix this!